src/frontend/js/components/RegisteredCreditCard/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { useMemo } from 'react';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import type * as Joanie from 'types/Joanie';
+
+const messages = defineMessages({
+  inputAriaLabel: {
+    defaultMessage: "{selected, select, true {Unselect} other {Select}} {title}'s card",
+    description: 'ARIA Label read by screen reader to inform which card is focused',
+    id: 'components.RegisteredCreditCard.inputAriaLabel',
+  },
+  expirationDate: {
+    defaultMessage: 'Expiration date: {expirationDate}',
+    description: 'Credit card expiration date label',
+    id: 'components.RegisteredCreditCard.expirationDate',
+  },
+});
+
+export interface Props extends Joanie.CreditCard {
+  selected: boolean;
+  handleSelect: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const RegisteredCreditCard = ({ selected, handleSelect, ...creditCard }: Props) => {
+  const intl = useIntl();
+
+  const expirationDate = useMemo(() => {
+    const { expiration_month: month, expiration_year: year } = creditCard;
+    return intl.formatDate(new Date(year, month - 1, 1), {
+      month: '2-digit',
+      year: 'numeric',
+    });
+  }, [creditCard.expiration_month, creditCard.expiration_year]);
+
+  const inputId = useMemo(() => `registered-credit-card-${creditCard.id}`, [creditCard.id]);
+
+  return (
+    <label className="registered-credit-card">
+      <p className="form-field">
+        <input
+          aria-label={intl.formatMessage(messages.inputAriaLabel, {
+            title: creditCard.title,
+            selected,
+          })}
```

Comment:
Maybe adding an `id` on the `registered-credit-card__infos` div and setting it as `aria-describedby` might be good here?

With a screen reader, when in a form, it's easy to miss info surrounding the form inputs if they are not actually tied to inputs in the HTML. Because in a form, you use a "form navigation" mode, mostly using the tab key, to move from input to input, contrary to the usual arrow key to move from any content to any content.

So here, all the credit card info can be easily missed. Adding the info in the `aria-describedby` could help. It would be announced as additional info, after the label.